vertical-pod-autoscaler/enhancements/7397-max-allowed-recommendation/README.md

```diff
@@ -0,0 +1,65 @@
+# KEP-7397: Max allowed recommendation configurable
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+   - [Goals](#goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+   - [Proposed Solution](#proposed-solution)
+   - [Alternatives](#alternatives)
+   - [Challenges](#challenges)
+   - [Test Plan](#test-plan)
+<!-- /toc -->
+
+## Summary
+VPA currently allows configuring the maximum allowed recommendation per resource using the field `.spec.resourcePolicy.containerPolicies[].maxAllowed`. This helps ensure that recommendations don't exceed the allocatable capacity of the largest node. Without this limit, a high recommendation could make the Pod unschedulable, causing downtime. We aim to improve the VPA recommender by making it aware of the maximum node size, preventing it from generating recommendations that would render Pods unschedulable.
+
+## Motivation
+In dynamic environments, manual configuration of .maxAllowed can be error-prone, especially as cluster sizes and node configurations change over time. Automating this process by making the VPA recommender aware of the largest node size ensures more reliable scaling decisions. This helps prevents situations where Pods become unschedulable due to excessive resource recommendations.
+
+### Goals
+* Make Recommendations More Accurate and Easier to Schedule: Try to have the VPA recommender suggest resources that fit within the largest node's capacity, helping pods get scheduled more easily and lowering the chance of downtime.
+* Adapt to Changing Environments: Adjust recommendations as the cluster size or node configurations change to better reflect the current state of the cluster.
+
+## Proposal
+We propose to improve the VPA recommender by making it aware of the maximum node size, preventing it from generating recommendations that would render Pods unschedulable. This will involve querying for cluster information and adjusting recommendations accordingly.
+
+## Design Details
+
+### Proposed Solution
+We propose to use the Cluster Autoscaler's provisioning-request API to obtain information about the maximum node size. This approach aligns well with CA's design and capabilities. The implementation will include:
```

Comment:
Provisioning requests answer the question of whether a particular set of pods can schedule or not (and potentially trigger scale-up in order to pre-create nodes for those pods). They don't directly provide information about underlying nodes.

Reply:
Would a `dryRun` mode on Provision Request solve this issue?

Reply:
I'm not much of an expert on Cluster Autoscaler (We're using Karpenter in our EKS clusters.) From that perspective, Karpenter provides a CRD called [NodePool](https://karpenter.sh/docs/concepts/nodepools), which might be useful. However, as far as I know, Karpenter doesn't support a `dryRun` method. I'm curious if CA offers any other features we could take advantage of.